app/models/user.rb

```diff
@@ -8,20 +8,20 @@ class User < ApplicationRecord
   private
 
   def password_validation
-    if !(has_uppercase?(password) && has_lowercase?(password) && has_number?(password))
-      errors.add :password, 'must have at least one uppercase letter, one lowercase letter and one number'
-    end
+    return if uppercase?(password) && lowercase?(password) && number?(password)
+
+    errors.add :password, 'must have at least one uppercase letter, one lowercase letter and one number'
   end
 
-  def has_uppercase?(password)
+  def uppercase?(password)
```

Comment:
not sure about this renaming, it kinda suggests now that it checks if the entire password is uppercase 🤔 whereas before it was very explicit that it is checking if the password _contains_ an uppercase character.

I don't have strong opinions on this but I preferred the previous naming.